sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -103,6 +104,8 @@ object DataType {
 
   def fromJson(json: String): DataType = parseDataType(parse(json))
 
+  def fromDdl(ddl: String): DataType = CatalystSqlParser.parseTableSchema(ddl)
```

Comment:
Also please add comments above the function to explain what is the purpose of these functions.